hardhat.config.ts

```diff
@@ -0,0 +1,140 @@
+import "@nomiclabs/hardhat-waffle";
+import "@nomiclabs/hardhat-ethers";
+import "@tenderly/hardhat-tenderly";
+import "@nomiclabs/hardhat-etherscan";
+import "@nomiclabs/hardhat-web3";
+import "hardhat-deploy";
+import "hardhat-deploy-ethers";
+import "@typechain/hardhat";
+
+import { resolve } from "path";
+import { config as dotenvConfig } from "dotenv";
+import { HardhatUserConfig } from "hardhat/config";
+import { NetworkUserConfig } from "hardhat/types";
+import { utils } from "ethers";
+import Web3 from "web3";
+
+dotenvConfig({ path: resolve(__dirname, "./.env") });
+
+const chainIds = {
+  ganache: 1337,
+  goerli: 5,
+  hardhat: 31337,
+  kovan: 42,
+  mainnet: 1,
+  rinkeby: 4,
+  ropsten: 3,
+  avalanche: 43114,
+  polygon: 137,
+};
+
+// Ensure that we have all the environment variables we need.
+const mnemonic = process.env.MNEMONIC;
+if (!mnemonic) {
+  throw new Error("Please set your MNEMONIC in a .env file");
+}
+
+const alchemyApiKey = process.env.ALCHEMY_API_KEY;
+if (!alchemyApiKey) {
+  throw new Error("Please set your ALCHEMY_ETH_API_KEY in a .env file");
+}
+
+const PRIVATE_KEY = process.env.PRIVATE_KEY;
+const ETHERSCAN_API_KEY = process.env.ETHERSCAN_API_KEY;
```

Comment:
For each network, we need a different API key. So, maybe let's add logics to use the required network key?